index.js

```diff
@@ -10,16 +10,27 @@ const fs = require('fs');
 
 const { createValuesObject, parseMetadataComments } = require('./lib/parser');
 const { checkKeys } = require('./lib/checker');
-const { combineMetadataAndValues, buildSectionsArrays, buildParamsToRenderList } = require('./lib/builder');
+const { combineMetadataAndValues, buildSectionsObjects, buildParamsToRenderList } = require('./lib/builder');
 const { insertReadmeTable, renderOpenAPISchema } = require('./lib/render');
 
-function getParameters(options) {
+/*
+* Parses the given values file for all metadata comments and returns an object containing
+* Returns an object with following structure:
+* {
+*   'parsedValues': [],              // contains all parameter related data
+*   'parsedSectionDescriptions': [], // contains key-value pairs of section name (key) and an array of lines for a description (value)
+* }
+*/
```

Comment:
IMHO the proper way of doing this would be to create a new data structure in a new `section.js` containing a class that we can use to create new sections with the `name`, `description` and `parameters` array.
The list of `parsedValues` could contain the `section` elements as well as `parameters` elements.
That way, it will be easier to manage during the whole code workflow

Reply:
@miguelaeh This was in fact my first approach but I had issues to distinguish parameter and section classes within the array so I went with this way to move forward. You are right that having another data structure is much cleaner.
I'll give it another try now, after a week not looking at the code.